src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/subquery/rewriter/SubqueryAliasRewriterTest.java

```diff
@@ -0,0 +1,135 @@
+                sqlString(
+                        rewrite(parse(
+                                "SELECT A.* FROM TbA as A WHERE A.a IN (SELECT B.b FROM TbB as B) and A.c > 10"))));
+    }
+
+    @Test
+    public void testOuterWithoutAliasInnerWithAlias() {
+        assertEquals(
+                sqlString(
+                        parse(
+                                "SELECT TbA_0.* FROM TbA as TbA_0 WHERE TbA_0.a IN (SELECT TbB.b FROM TbB as TbB) and TbA_0.c > 10")),
+                sqlString(
+                        rewrite(parse(
+                                "SELECT * FROM TbA WHERE a IN (SELECT TbB.b FROM TbB as TbB) and c > 10"))));
+    }
+
+    @Test
+    public void testOuterWithoutAliasInnerMixAlias() {
+        String expect = "SELECT TbA_0.* FROM TbA as TbA_0 WHERE TbA_0.a IN (SELECT B.b FROM TbB as B) and TbA_0.c > 10";
+
+        assertEquals(
+                sqlString(
+                        parse(
+                                expect)),
+                sqlString(
+                        rewrite(parse(
+                                "SELECT * FROM TbA WHERE a IN (SELECT b FROM TbB as B) and c > 10"))));
+
+        assertEquals(
+                sqlString(
+                        parse(
+                                expect)),
+                sqlString(
+                        rewrite(parse(
+                                "SELECT * FROM TbA WHERE a IN (SELECT TbB.b FROM TbB as B) and c > 10"))));
+    }
+
+    @Test
+    public void testOuterWithAliasInnerWithoutAlias() {
+        assertEquals(
+                sqlString(
+                        parse(
+                                "SELECT TbA.* FROM TbA as TbA WHERE TbA.a IN (SELECT TbB_0.b FROM TbB as TbB_0) and TbA.c > 10")),
+                sqlString(
+                        rewrite(parse(
+                                "SELECT TbA.* FROM TbA as TbA WHERE TbA.a IN (SELECT b FROM TbB ) and TbA.c > 10"))));
+    }
+
+    @Test
+    public void testOuterMixAliasInnerWithoutAlias() {
+        String expect = "SELECT A.* FROM TbA as A WHERE A.a IN (SELECT TbB_0.b FROM TbB as TbB_0) and A.c > 10";
+
+        assertEquals(
+                sqlString(
+                        parse(
+                                expect)),
+                sqlString(rewrite(
+                        parse(
+                                "SELECT TbA.* FROM TbA as A WHERE a IN (SELECT b FROM TbB ) and TbA.c > 10"))));
+
+        assertEquals(
+                sqlString(
+                        parse(
+                                expect)),
+                sqlString(
+                        rewrite(parse(
+                                "SELECT * FROM TbA as A WHERE TbA.a IN (SELECT b FROM TbB ) and TbA.c > 10"))));
+    }
+
+
+    private String sqlString(SQLExpr expr) {
+        return SQLUtils.toMySqlString(expr);
+    }
+
+    private SQLQueryExpr parse(String sql) {
```

Comment:
same comment - please reuse some existing method for parsing string to sql instead of repeating it in every test

Reply:
Done.